src/acc.rs

```diff
@@ -330,10 +678,22 @@ fn intra(intra: &IntracellularProperties) -> Result<Vec<Decor>> {
                 initialConcentration,
                 initialExtConcentration,
                 segmentGroup,
+                concentrationModel,
                 ..
             }) => {
                 result.push(Decor::xi(segmentGroup, ion, initialConcentration));
                 result.push(Decor::xo(segmentGroup, ion, initialExtConcentration));
+                result.push(Decor::new(
+                    segmentGroup,
+                    Paintable::Mech(
```

Comment:
I think I added some helpers to `Decor` for constructing all these, check if there's something that fits your needs here.